### PR TITLE
[docs] Document Rust UDF development workflow

### PR DESCRIPTION
**Pull request by ryzhyk**
See commit messages.

### CODE REVIEW

File: docs/sql/udf.md [hunk 1]
```diff
@@ -11,106 +11,266 @@ The following example shows a user-defined function:
...
-Rust user-defined functions.
+Feldera supports UDFs implemented in Rust. To define a UDF, users declare the function
+in SQL without providing a body. The corresponding Rust implementation is supplied in a separate
+Rust file, which contains the logic for all UDFs declared within the SQL program. A complete
+set of files that defines the pipeline in this case is:
 
-The following example shows a user-defined function:
+| File          | Description                                                                                         |
+|---------------|-----------------------------------------------------------------------------------------------------|
+| `program.sql` | The SQL code, including table, view, and function declarations.                                     |
```
Comment (mihaibudiu) [inline docs/sql/udf.md:55]: and user-defined function declarations

File: docs/sql/udf.md [hunk 2]
```diff
@@ -11,106 +11,266 @@ The following example shows a user-defined function:
...
+| `stubs.rs`    | Auto-generated by the SQL compiler; contains Rust stubs for UDFs ([see below](#step-2-locate-the-auto-generated-rust-prototype)).                       |
+| `udf.rs`      | User-provided implementation of UDFs in Rust.                                                       |
+| `udf.toml`    | List of external crates required by the code in `udf.rs` in the `Cargo.toml` format.                |
+
+We recommend using the Feldera WebConsole to develop and debug Rust UDFs.
```
Comment (mihaibudiu) [inline docs/sql/udf.md:60]: link to webconsole?

File: docs/sql/udf.md [hunk 3]
```diff
@@ -11,106 +11,266 @@ The following example shows a user-defined function:
...
+curl -i -X PUT http://localhost:8080/v0/pipelines/udf_api_test \
+-H 'Content-Type: application/json' \
+-d "$(jq -Rsn \
+  --rawfile sql program.sql \
+  --rawfile udf_rust udf.rs \
+  --rawfile udf_toml udf.toml \
+  '{
+    name: "udf_api_test",
+    description: "Create a UDF using REST API",
+    program_code: $sql,
```
Comment (mihaibudiu) [inline docs/sql/udf.md:234]: where do these variables come from?

File: docs/sql/udf.md [hunk 4]
```diff
@@ -11,106 +11,266 @@ The following example shows a user-defined function:
...
+Rust file, which contains the logic for all UDFs declared within the SQL program. A complete
+set of files that defines the pipeline in this case is:
 
-The following example shows a user-defined function:
+| File          | Description                                                                                         |
+|---------------|-----------------------------------------------------------------------------------------------------|
+| `program.sql` | The SQL code, including table, view, and function declarations.                                     |
+| `stubs.rs`    | Auto-generated by the SQL compiler; contains Rust stubs for UDFs ([see below](#step-2-locate-the-auto-generated-rust-prototype)).                       |
+| `udf.rs`      | User-provided implementation of UDFs in Rust.                                                       |
+| `udf.toml`    | List of external crates required by the code in `udf.rs` in the `Cargo.toml` format.                |
```
Comment (snkas) [inline docs/sql/udf.md:58]: User-provided list of external crates required by the code in `udf.rs` in the same format as that within the `[dependencies]` section of the well-known `Cargo.toml` format.

File: sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/CompilerMain.java [hunk 5]
```diff
@@ -222,10 +222,11 @@ CompilerMessages run() throws SQLException {
                     System.out.println("Writing UDF stubs to file " + stubs);
                 protosStream.append("""
 // Compiler-generated file
-// Stubs for user-defined functions
+// This file contains stubs for user-defined functions. Copy these stubs to
```
Comment (snkas) [inline sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/CompilerMain.java:225]: I am not as familiar with the word "stub", could we maybe add "function signature" somewhere, like:

```
// This file contains stubs for the user-defined functions from the SQL program.
// Each stub defines a function name and corresponding signature that must be implemented in `udf.rs`.
// Copy these stubs to `udf.rs` and replace their bodies with the actual implementations.
// For more detailed documentation: https://docs.feldera.com/sql/udf
```

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/MetadataTests.java [hunk 6]
```diff
@@ -334,10 +334,11 @@ pub fn EMPTY() -> Result<Option<String>, Box<dyn std::error::Error>> {
         List<String> str = Files.readAllLines(protos);
         Assert.assertEquals("""
                 // Compiler-generated file
-                // Stubs for user-defined functions
+                // This file contains stubs for user-defined functions. Copy these stubs to
+                // `udf.rs`, replacing their bodies with the actual UDF implementations.
+                // See detailed documentation in https://docs.feldera.com/sql/udf.
```
Comment (snkas) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/MetadataTests.java:339]: (see other comment)

**Review by snkas — APPROVED** [review]
The additional documentation helps a lot! Looks good.

File: docs/sql/udf.md [hunk 7]
```diff
@@ -11,106 +11,266 @@ The following example shows a user-defined function:
...
+
+We recommend using the Feldera WebConsole to develop and debug Rust UDFs.
+The WebConsole code editor has a separate tab for each of these files.
```
[the remark this answers is not in the document]
Comment (snkas) [reply docs/sql/udf.md:61]: *function prototypes

File: docs/sql/udf.md (hunk 2 above)
Comment (ryzhyk) [reply docs/sql/udf.md:60]: we don't have webconsole docs yet

File: docs/sql/udf.md (hunk 3 above)
Comment (ryzhyk) [reply docs/sql/udf.md:234]: yep

**Pull request merged**